File: DataStructure/Python/10828.py
class stack:
    value = [0 for i in range(10001)]
    size = 0
    
    def isEmpty(self):
        if self.size == 0: return True
        else: return False
        
    def push(self, x):
        self.value[self.size] = x
        self.size += 1
        
    def pop(self):
        if self.isEmpty():
            print(-1)
        else:
            print(self.value[self.size-1])
            self.size -= 1
            
    def top(self):
        if self.isEmpty():
            print(-1)
        else:
            print(self.value[self.size-1])
            
    def getSize(self):
        print(self.size)
    
from sys import stdin
n = int(stdin.readline())

s = stack()

for i in range(n):
    cmd = stdin.readline().split()
    if cmd[0].startswith('push'):
        s.push(int(cmd[1]))
    elif cmd[0].startswith('pop'):
        s.pop()
    elif cmd[0].startswith('top'):
        s.top()
    elif cmd[0].startswith('size'):
        s.getSize()
    elif cmd[0].startswith('empty'):
        if s.isEmpty(): print(1)
        else: print(0)

File: BaekJoon/DataStructure/C++/10828.cpp
//2021.07.27
#include<iostream>
using namespace std;

class  stack{
    int value[10001];
    int size = 0;

    public:
    bool isEmpty(){
        if(size == 0){
            return true;
        }else {
            return false;
        }
    }

    void push(int x){
        value[size] = x;
        size++;
    }

    void pop(){
        if(isEmpty()){
            cout << -1 << '\n';
        }else{
            cout << value[size-1] << '\n';
            size--;
        }
    }

    void getSize(){
        cout << size << '\n';
    }

    void top(){
        if(isEmpty()){
            cout << -1 << '\n';
        }else{
            cout << value[size-1] << '\n';
        }
    }
};


int main(){
    int n;
    stack s;
    cin >> n;
    cin.ignore();
    for(int i = 0; i < n; i++){
        string command;
        cin >> command;
        if(command == "push"){
            int x;
            cin >> x;
            s.push(x);
        }else if(command == "pop"){
            s.pop();
        }else if(command == "size"){
            s.getSize();
        }else if(command == "empty"){
            if(s.isEmpty()) cout << 1 << '\n';
            else cout << 0 << '\n';
        }else if(command == "top"){
            s.top();
        }else{}
        

    }
    return 0;
}